src/mono/mono/mini/mini-arm64.c

```diff
@@ -30,6 +30,46 @@
 
 #include "interp/interp.h"
 
+// The following defines are here to support the inclusion of simd-arm64.h
+#define EXPAND(x) x
+#define PARENTHESIZE(...) (__VA_ARGS__)
+#define EXPAND_FUN(m, ...) EXPAND(m PARENTHESIZE(__VA_ARGS__))
+
+#define OPFMT_DS dreg, sreg1
+#define OPFMT_WDS _w, dreg, sreg1
+#define OPFMT_WTDS _w, _t, dreg, sreg1
+#define OPFMT_WTDSS _w, _t, dreg, sreg1, sreg2
+#define OPFMT_WTDSS_REV _w, _t, dreg, sreg2, sreg1
+#define OPFMT_WTDI(imm) _w, _t, dreg, (imm)
+#define OPFMT_WDDD _w, dreg, dreg, dreg
+#define _UNDEF(...) g_assert_not_reached ()
+#define SIMD_OP_CODE(reg_w, op, c) ((reg_w << 31) | (op) << 16 | (c))
+#define VREG_64 VREG_LOW
+#define VREG_128 VREG_FULL
+#define OPCODE_BASIC 0
+#define OPCODE_SIMD 1
+#define OPCODE_SIMD_EX 2
+
+#define SIMD_OP_INTERNAL(code, reg_w, fmt, i8fun, i16fun, i32fun, i64fun, f32fun, f64fun) \
+	if (_f) { \
```

Comment:
Checking if type is float seems to be unnecessary here. `_t` has a clear one-to-one mapping here.